README.md

```diff
@@ -11,3 +11,157 @@ Caching, aggregation, and relaying for xDS compliant clients and origin servers
 To get started:
 
 * [Contributing guide](CONTRIBUTING.md)
+
+## Design
+
+## Aggregation Rules
+
+xds-relay uses aggregation rules to determine how to aggregate inbound connections to the origin
+management server ([example](https://github.com/envoyproxy/xds-relay/blob/c83a956952c3e9762a08420ffc1c6bda5e0fff04/integration/testdata/keyer_configuration_e2e.yaml)
+).
+
+xDS requests that match under the same set of aggregation rules will fall into the same cache
+bucket. The cache is a mapping of a aggregation key to a xDS [DiscoveryResponse](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/discovery.proto#discoveryresponse)
+. This implies that all inbound xDS requests that map to the same aggregation key will retrieve the
+same DiscoveryResponse. This is useful in multi-cluster scenarios where xDS configuration does not
+differ greatly between clusters. For example, a service application scaled across 100 hosts will
+likely require the same data plane configurations across all hosts.
+
+At a high level, an aggregation rule is formed from `fragments`, `rules`, `match`, and `result`s.
+
+#### Fragments
+A final aggregation key may look like `fooservice_production_lds`. Each of those identifiers
+separated by `_` is what we call a fragment. _fooservice_, _production_, and _lds_ are 3
+fragments in the previous example.
+
+#### Rules
+Each `fragment` can have multiple `rule`s ([example](https://github.com/envoyproxy/xds-relay/blob/c83a956952c3e9762a08420ffc1c6bda5e0fff04/integration/testdata/keyer_configuration_e2e.yaml#L28-L52)
+). A rule defines how to generate a fragment from an xDS [DiscoveryRequest](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/discovery.proto#discoveryrequest)
+. The first matching rule applies, and the rest are ignored.
```

Comment:
I don't think this is true, i.e. we can have multiple matches in a list of rules. The mapper currently [matches](https://github.com/envoyproxy/xds-relay/blob/c83a956952c3e9762a08420ffc1c6bda5e0fff04/internal/app/mapper/mapper.go#L49-L66) fragments for all rules, not only the first one (as [this comment](https://github.com/envoyproxy/xds-relay/blob/master/api/protos/aggregation/v1/aggregation.proto#L27-L28) implies).